File: programmers/3진법 뒤집기.py
def solution(n):
    answer = 0
    result = ''
    while n > 0:
        result = str(n % 3) + result # 나머지로 3진법 표현
        n = n // 3
    print(result)
    for i in range(len(result)):
        answer += int(result[i]) * 3**i
    return answer

File: programmers_level1/3진법_뒤집기.cpp
#include <vector>

using namespace std;

int solution(int n) {
    int answer{ 0 };
    vector<int> ternary;

    int quotient{ n }, remainder{ 0 };
    while (quotient > 0) {
        remainder = quotient % 3;
        quotient /= 3;
        ternary.push_back(remainder); // 역순으로 반전된 값 저장
    }
    int mul{ 1 }; // for loop에서 multiple variable initialize하려면 복잡함
    for (int i=ternary.size()-1; i>=0; --i) { // 뒤의 자리에서부터 계산
        answer += ternary[i]*mul;
        mul *= 3;
    }

    return answer;
}